vermouth/rcsu/go_vs_includes.py

```diff
@@ -0,0 +1,147 @@
+        Each virtual site copies the resid, resname, and chain of the backbone
+        bead. It also copies the *reference* to the position array, so the virtual
+        site position follows if the backbone bead is translated. The virtual sites
+        are added *after* all the other atoms of the molecule, each in its own
+        charge group, with "CA" as atomname, and a charge of 0. The atomname and
+        charge can be set with the `atomname` and `charge` argument, respectively.
+
+        The bead type of the virtual sites is names "<prefix>_<resid>". Where
+        `prefix` is provided as an argument of the function, and is expected to be
+        the molecule type name.
+
+        Parameters
+        ----------
+        molecule: vermouth.molecule.Molecule
+            The molecule to augment with virtual sites.
+        prefix: str
+            The prefix to use for bead type names. Usually the molecule type name.
+        backbone: str
+            The atomname of the backbone beads.
+        atomname: str
+            The atomname of the virtual sites.
+        charge: float or int
+            The charge of the virtual sites.
+        """
+        # If there are no atoms, then there is nothing to do. We can exit early,
+        # avoiding to deal with empty iterators.
+        if not molecule.nodes:
+            return
+        virtual_site_nodes = []
+        virtual_sites = []
+        atomtypes = {}
+        new_node_id = max(molecule.nodes)
+        charge_groups = nx.get_node_attributes(molecule, 'charge_group').values()
+        new_charge_group = max(charge_groups) if charge_groups else 0
+        for node_id, atom in molecule.nodes(data=True):
+            if atom.get('atomname') == backbone:
+                new_node_id += 1
+                new_charge_group += 1
+
+                virtual_site_nodes.append((new_node_id, {
+                    'resid': atom['resid'],
+                    '_old_resid': atom['_old_resid'],
+                    'resname': atom['resname'],
+                    'atype': '{}_{}'.format(prefix, atom['resid']),
+                    'charge_group': new_charge_group,
+                    'chain': atom['chain'],
+                    'position': atom['position'],
+                    'atomname': atomname,
+                    'charge': charge,
+                    'mass': 0.0,
+                    'cgsecstruct': atom.get('cgsecstruct', None),
+                }))
+                virtual_sites.append(Interaction(
+                    atoms=[new_node_id, node_id],
+                    parameters=['1'],
+                    meta={'go_vs': True, 'group': 'Virtual go site'},
+                ))
+            vs_params = Atomtype(node=new_node_id, molecule=molecule, sigma=0.0, epsilon=0.0, meta={})
+            self.system.gmx_topology_params['atomtypes'].append(vs_params)
+
+        molecule.add_nodes_from(virtual_site_nodes)
```

Comment:
Do we want to add edges between the VS and the BB to keep everything connected?

Reply:
somehow the residue graph making seems to work even if that doesn't happen, but I think we would want to